3scaleAdapter/threescale_test.go

```diff
@@ -13,18 +13,49 @@ import (
 	"testing"
 )
 
+func Test_HandleAuthorization(t *testing.T) {
+
+	var handleAuthorizationTests = []struct {
+		httpStatus     int
+		expectedStatus int32
+	}{
+		{http.StatusOK, 0},
+		{http.StatusAccepted, 0},
+		{http.StatusRequestTimeout, 7},
+		{http.StatusBadGateway, 7},
+		{http.StatusGatewayTimeout, 7},
+		{http.StatusForbidden, 7},
+	}
+
+	ctx := context.TODO()
 
+	r := &authorization.HandleAuthorizationRequest{
+		Instance: &authorization.InstanceMsg{
+			Subject: &authorization.SubjectMsg{},
+			Action:  &authorization.ActionMsg{},
+		},
+		AdapterConfig: &types.Any{},
+		DedupId:       "",
+	}
 
-var handleAuthorizationTests = []struct {
-	httpStatus  int
-	expectedStatus int32
-}{
-	{http.StatusOK, 0},
-	{http.StatusAccepted, 0},
-	{http.StatusRequestTimeout, 7},
-	{http.StatusBadGateway, 7},
-	{http.StatusGatewayTimeout, 7},
-	{http.StatusForbidden, 7},
+	for _, tt := range handleAuthorizationTests {
+		ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.WriteHeader(tt.httpStatus)
+		}))
+		l, _ := net.Listen("tcp", "127.0.0.1:8090")
+		ts.Listener = l
+		ts.Start()
+
```

Comment:
@jmprusi it's usually good practice to call `defer ts.Close()` in cases like this directly after starting, opening etc so we ensure it always gets called.

Reply:
`defer` executes when the function is returned. we are inside a for loop here so `defer` is not recommended and will leak.

Reply:
Yes you are correct. Need ☕️

Reply:
☕️